### PR TITLE
JDK11+ nashorn scripting, SecurityManager update, Gradle 7.6

### PR DESCRIPTION
**Pull request by tcnh**
For reference. 

This branch runs scripting (slim symbol expressions) with Java 11 and up. It incorporates OpenJDK Nashorn instead of depending on the previously built in nashorn scriptengine.
Also removed the no-longer existed methods from the SecurityManager implementation (but it should be removed entirely imho) and upgraded Gradle and other dependencies.

Note: this branch can no longer be compiled using JDK8.

### CODE REVIEW

File: buildSrc/src/main/groovy/LessCompiler.groovy [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
+import org.gradle.api.tasks.JavaExec
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
 
 class LessCompiler extends JavaExec {
   @InputDirectory
   def inputDir
 
-  @Input
+  @OutputFile
```
Comment (fhoeben) [inline buildSrc/src/main/groovy/LessCompiler.groovy:10]: This seems like a strange change. Is this file suddenly no longer an input but an output?

File: buildSrc/src/main/groovy/WikiFileListBuilderTask.groovy [hunk 2]
```diff
@@ -12,7 +12,7 @@ public class WikiFileListBuilderTask extends DefaultTask {
   @OutputFile
   File updateDoNotCopyOverListFile
 
-  @Input
+  @OutputFiles
```
Comment (fhoeben) [inline buildSrc/src/main/groovy/WikiFileListBuilderTask.groovy:15]: idem

File: src/fitnesse/slim/SlimExpressionEvaluator.java [hunk 3]
```diff
@@ -14,21 +14,13 @@
...
   public SlimExpressionEvaluator() {
-    this("JavaScript");
+    this.engine = new NashornScriptEngineFactory().getScriptEngine();
   }
 
-  public SlimExpressionEvaluator(String engineName) {
-    this(ENGINE_MANAGER.getEngineByName(engineName));
-  }
-
-  public SlimExpressionEvaluator(ScriptEngine engine) {
```
Comment (fhoeben) [inline src/fitnesse/slim/SlimExpressionEvaluator.java:29]: Why not keep this constructor and feed it the nashhorn engine?

**Review by fhoeben — COMMENTED** [review]
Looks pretty good.
If you are happy with it I think it might be mergeable once the release notes page is also updated (we'll also need to update the FitNesse.org site as well of course). 
Do you have any open issues that should be addressed?

**Comment by fhoeben** [comment]
@jediwhale what do you think about switching to Java 11 and dropping Java 8 support. Is that an issue for you?

**Comment by jediwhale** [comment]
OK with me

File: buildSrc/src/main/groovy/LessCompiler.groovy (hunk 1 above)
Comment (tcnh) [reply buildSrc/src/main/groovy/LessCompiler.groovy]: yes, that was a leftover from some debugging. reverted

File: src/fitnesse/slim/SlimExpressionEvaluator.java (hunk 3 above)
Comment (tcnh) [reply src/fitnesse/slim/SlimExpressionEvaluator.java:29]: I thought to do it this way, so we had the nashorn import and scriptengine initialization only in this file. I can see the compatibility argument to keep the old constructor. Not really a preference for one or the other.

**Pull request merged**